http2/README.md

```diff
@@ -11,107 +11,80 @@ Note: due to the lack of H2C support in popular frameworks, the Ruby and Python
-> git clone https://github.com/SAP-samples/cf-routing-samples
-> cd cf-routing-samples/http2
-```
-
-Build the app (Java apps only)
-
-```shell
-> ./gradlew build
-```
-
-Choose a name and push the application without adding a route
-
-```shell
-> cf push --no-route http2-example-app-go
-
-# For the Java app, specify the path to the distribution zip
-> cf push --no-route http2-example-app-java --buildpack java_buildpack --path app/build/distributions/app.zip
+# edit the vars.yml file and replace <domain.for.app.routes> with required domain
+# my.cf.app.domain is used as an example for demonstration
+> vi vars.yml
+> cat vars.yml
+domain: my.cf.app.domain
 ```
 
-Map an HTTP2 route with the application
-
+#### For java app do the following
 ```shell
-> export CF_APPS_DOMAIN=my-apps.cf.example.com
-> cf map-route http2-example-app-go $CF_APPS_DOMAIN --hostname http2-example-app-go --destination-protocol http2
+# please make sure that the openjdk@11 is installed and gradle wrapper is pointing to it
+> ./gradlew clean build
 ```
-
-Check the app is working
-
+### Deploy all apps
 ```shell
-> curl https://http2-example-app-go.$CF_APPS_DOMAIN /
-Hello! This Go application is speaking plain text HTTP2 (H2C) with the CF routing layer
+# run the following commands in cf environment
+> cd http2
+> cf push --manifest apps-manifest.yml --vars-file vars.yml --vars-file gradle.properties
 ```
-
-### How to deploy gRPC apps
-
-Make sure you have [CF CLI](https://docs.cloudfoundry.org/cf-cli/install-go-cli.html) version 8 or higher installed
+## Deploying apps individually 
+* Make sure you have [CF CLI](https://docs.cloudfoundry.org/cf-cli/install-go-cli.html) version 8 or higher installed
 
 ```shell
 > cf version
 cf version 8.0.0+e8d41cf8e.2021-09-16
 ```
-
-Build the app (Java apps only)
-
-```shell
-> ./gradlew build
-```
-
-Choose a name and push the application without adding a route
-
-```shell
-> cf push --no-route grpc-example-app-go
-
-# For the Java app, specify the path to the distribution zip
-> cf push --no-route grpc-example-app-java --buildpack java_buildpack --path app/build/distributions/app.zip
-```
-
-Map an HTTP2 route with the application. It must be a route on an apps domain supporting mutual TLS (mTLS) and you must have the client certificate and private key available on your machine
-
+* And set the desired domain in [vars.yml](vars.yml) file
+### How to deploy H2C apps
+* Go: [go-http2](go-http2)
+* Java: [java-http2](java-http2) (Note: refer [pre-req](#For java app do the following) for Java)
+* Node: [node-http2](node-http2)
+* Python: [python-http2](python-http2)
+* Ruby: [ruby-http2](ruby-http2)
+#### Check H2C apps
 ```shell
-> export CF_MTLS_APPS_DOMAIN=my-mtls-apps.cf.example.com
-> cf map-route grpc-example-app-go $CF_MTLS_APPS_DOMAIN --hostname grpc-example-app-go --destination-protocol http2
+# format: https://<language>-http2-test.my.cf.app.domain
+> curl -v --http2-prior-knowledge https://go-http2-test.my.cf.app.domain
```

Comment:
For easier copy&paste execution, you could also do:
```shell
export DOMAIN=cf.domain.example.com
curl -v --http2-prior-knowledge "https://go-http2-test.$DOMAIN"
[...]
```

The `>` in front of each line will make someone redirect something into a file they don't want. People _will_ copy&paste this ;-)